src/cephadm/cephadm

```diff
@@ -3189,6 +3214,73 @@ class CephContainer:
         self.memory_request = memory_request
         self.memory_limit = memory_limit
 
+    @classmethod
+    def for_daemon(cls,
```

Comment:
I don't entirely understand this for_daemon method. From what I can tell it just returns a CephContainer instance where it builds the cname out of a given daemon_id, daemon_type and fsid rather than being given the cname directly as well as pulling the image from the context rather than just being given the image. What is the purpose of having this rather than just a normal CephContainer instance where we pass `fsid.daemon_type.daemon_id` as the cname and `ctx.image` as the image?

Reply:
the container name generation was used multiple times. I used `for_daemon` to have a single way to define the name of the container. Helped me to do this PR. You think I should remove it?

Reply:
No, I guess it just felt like a lot of lines to get around manually building the container name, but it does make sense to have some defined way of doing this. What do you think about just a general make_name function to use when passing cname in (cname = make_cname(fsid, daemon_type, daemon_id))? If you don't like it I think for_daemon can stay, because I agree we should have some functionality to build these names instead of doing it manually over and over.